### ISSUE TITLE
Same target built twice? How is this possible?

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#2144]
I'm looking at a build log file where TargetStarted is called twice, the project and target names are the same, and the only difference is targetId and start time:
![image](https://cloud.githubusercontent.com/assets/679326/26533277/a4a1d802-43cb-11e7-919e-394b465443d1.png)

Is this normal/expected?

**Comment by AndyGerlicher** [comment]
Team Triage:
We'd need more information or ideally a minimal repro to figure this out. Do you have a way to get that?

**Comment by KirillOsenkov** [comment]
Well, this is during the build of Roslyn and the full log is here:
https://www.dropbox.com/s/9sa6dgi31m6sb1d/roslyn.binlog?dl=0

This build happens nightly so I'll see if it happens again.

But, to confirm, this is not expected behavior, even with cross-targeting and such? Note that these are coming from different threads and slightly different time (2 seconds apart), but the same node, submission, etc.

Also note that the targets in question are relatively new (added recently for cross-targeting?). Maybe CallTarget can result in this?

**Comment by atanasa** [comment]
I have an isolated repro for the case of 2 target frameworks - the project+target is build twice (once for each target framework) even though MSBuild task does not include the TargetFramework as input to the project. Note that the project being built twice is a plain MSBuild project (not a C# project) with a custom target. I would expect that MSBuild handles these in a generic way.
This is the sample solution that reproduces this:
[ReproDoubleBuild.zip](https://github.com/Microsoft/msbuild/files/1108507/ReproDoubleBuild.zip)

**Comment by KirillOsenkov** [comment]
I think this is because this target does batching:
https://github.com/Microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L1525-L1526

So it ends up getting run three times:
![image](https://user-images.githubusercontent.com/679326/27851721-ee8331ac-6110-11e7-904f-cc1ea56276ac.png)

**Comment by KirillOsenkov** [comment]
cc @nguerrera is this the expected behavior? (just checking)

**Comment by nguerrera** [comment]
Yes. Target batching is expected if that's how it's represented in the log.
